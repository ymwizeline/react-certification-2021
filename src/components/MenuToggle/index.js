export { default } from './MenuToggle.component';
