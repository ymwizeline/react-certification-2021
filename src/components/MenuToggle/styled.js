import styled from 'styled-components';

export const Container = styled.button`
  margin: 10px;
  cursor: pointer;
`;

export const Line = styled.span`
  display: block;
  width: 22px;
  height: 3px;
  margin-bottom: 4px;
  margin-top: 4px;
  position: relative;
  border-radius: 3px;
  background: ${(props) => props.theme.secondary};
`;
