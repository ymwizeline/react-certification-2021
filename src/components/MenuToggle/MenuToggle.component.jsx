import React from 'react';
import { Container, Line } from './styled';

const MenuToggle = () => {
  return (
    <Container as="div">
      <Line />
      <Line />
      <Line />
    </Container>
  );
};

export default MenuToggle;
