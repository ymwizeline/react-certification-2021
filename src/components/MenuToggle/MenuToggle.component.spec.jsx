import React from 'react';
import renderer from 'react-test-renderer';
import MenuToggle from './MenuToggle.component';
import 'jest-styled-components';

describe('MenuToggle component', () => {
  it(`renders properly`, () => {
    const element = renderer.create(<MenuToggle />).toJSON();

    expect(element).toMatchSnapshot();
  });
});
