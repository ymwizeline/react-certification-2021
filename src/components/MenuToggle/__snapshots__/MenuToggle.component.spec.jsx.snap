// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuToggle component renders properly 1`] = `
.c0 {
  margin: 10px;
  cursor: pointer;
}

.c1 {
  display: block;
  width: 22px;
  height: 3px;
  margin-bottom: 4px;
  margin-top: 4px;
  position: relative;
  border-radius: 3px;
}

<div
  className="c0"
>
  <span
    className="c1"
  />
  <span
    className="c1"
  />
  <span
    className="c1"
  />
</div>
`;
